packages/authentication/src/decorators/authenticate.decorator.ts

```diff
@@ -4,50 +4,102 @@
 // License text available at https://opensource.org/licenses/MIT
 
 import {
+  ClassDecoratorFactory,
   Constructor,
+  DecoratorFactory,
   MetadataInspector,
   MethodDecoratorFactory,
 } from '@loopback/context';
-import {AUTHENTICATION_METADATA_KEY} from '../keys';
+import {
+  AUTHENTICATION_METADATA_CLASS_KEY,
+  AUTHENTICATION_METADATA_KEY,
+  AUTHENTICATION_METADATA_METHOD_KEY,
+} from '../keys';
+import {AuthenticationMetadata} from '../types';
 
-/**
- * Authentication metadata stored via Reflection API
- */
-export interface AuthenticationMetadata {
-  strategy: string;
-  options?: object;
-}
+class AuthenticateClassDecoratorFactory extends ClassDecoratorFactory<
+  AuthenticationMetadata
+> {}
 
 /**
  * Mark a controller method as requiring authenticated user.
  *
- * @param strategyName - The name of the authentication strategy to use.
+ * @param strategyNameOrMetadata - The name of the authentication strategy to use
+ * or the authentication metadata object.
  * @param options - Additional options to configure the authentication.
  */
-export function authenticate(strategyName: string, options?: object) {
-  return MethodDecoratorFactory.createDecorator<AuthenticationMetadata>(
-    AUTHENTICATION_METADATA_KEY,
-    {
-      strategy: strategyName,
-      options: options || {},
-    },
-    {decoratorName: '@authenticate'},
-  );
+export function authenticate(
+  strategyNameOrMetadata: string | AuthenticationMetadata,
+  options?: object,
+) {
+  return function authenticateDecoratorForClassOrMethod(
+    // Class or a prototype
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    target: any,
+    method?: string,
+    // Use `any` to for `TypedPropertyDescriptor`
+    // See https://github.com/strongloop/loopback-next/pull/2704
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    methodDescriptor?: TypedPropertyDescriptor<any>,
+  ) {
+    let spec: AuthenticationMetadata;
+    if (typeof strategyNameOrMetadata === 'object') {
+      spec = strategyNameOrMetadata;
+    } else {
+      spec = {strategy: strategyNameOrMetadata, options: options || {}};
+    }
+    if (method && methodDescriptor) {
+      // Method
+      return MethodDecoratorFactory.createDecorator<AuthenticationMetadata>(
+        AUTHENTICATION_METADATA_KEY,
+        spec,
+        {decoratorName: '@authenticate'},
+      )(target, method, methodDescriptor);
+    }
+    if (typeof target === 'function' && !method && !methodDescriptor) {
+      // Class
+      return AuthenticateClassDecoratorFactory.createDecorator(
+        AUTHENTICATION_METADATA_CLASS_KEY,
+        spec,
+        {decoratorName: '@authorize'},
```

Comment:
`@authenticate` instead of `@authorize`?

Reply:
Good catch